src/containers/PerfherderGraphContainer/index.jsx

```diff
@@ -23,25 +24,45 @@ const styles = () => ({
 class PerfherderGraphContainer extends Component {
   state = {
     data: null,
+    tooltipModel: null,
+    canvas: null,
     isLoading: false,
   };
 
   async componentDidMount() {
-    await this.fetchSetData(this.props);
+    this.fetchSetData(this.props);
   }
 
   async fetchSetData({ series }) {
     try {
       this.setState({ isLoading: true });
       this.setState(await getPerfherderData(series));
     } finally {
-      this.setState({ isLoading: false });
+      const self = this;
+
+      this.setState(prevState => {
+        const { options } = prevState;
+
+        options.tooltips.custom = function custom(tooltipModel) {
```

Comment:
The `options` is undefined, and this method throws an error.  Please add a check for that case.

Reply:
https://deploy-preview-361--firefox-health-dashboard.netlify.com/android/tp6m?test=loadtime&platform=android-hw-g5-7-0-arm7-api-16